pkg/microservice/aslan/core/system/service/registry.go

```diff
@@ -199,44 +216,293 @@ func ListAllRepos(log *zap.SugaredLogger) ([]*RepoInfo, error) {
 	return repoInfos, nil
 }
 
+type imageReqJob struct {
+	regService   registry.Service
+	endPoint     registry.Endpoint
+	registryInfo *models.RegistryNamespace
+	repoImages   *registry.Repo
+	tag          string
+	failedTimes  int
+}
+
+type writeStatus struct {
+	m      *sync.Mutex
+	status *int
+}
+
 func ListReposTags(registryInfo *commonmodels.RegistryNamespace, names []string, logger *zap.SugaredLogger) ([]*RepoImgResp, error) {
 	var regService registry.Service
 	if registryInfo.AdvancedSetting != nil {
 		regService = registry.NewV2Service(registryInfo.RegProvider, registryInfo.AdvancedSetting.TLSEnabled, registryInfo.AdvancedSetting.TLSCert)
 	} else {
 		regService = registry.NewV2Service(registryInfo.RegProvider, true, "")
 	}
+
+	endPoint := registry.Endpoint{
+		Addr:      registryInfo.RegAddr,
+		Ak:        registryInfo.AccessKey,
+		Sk:        registryInfo.SecretKey,
+		Namespace: registryInfo.Namespace,
+		Region:    registryInfo.Region,
+	}
 	repos, err := regService.ListRepoImages(registry.ListRepoImagesOption{
-		Endpoint: registry.Endpoint{
-			Addr:      registryInfo.RegAddr,
-			Ak:        registryInfo.AccessKey,
-			Sk:        registryInfo.SecretKey,
-			Namespace: registryInfo.Namespace,
-			Region:    registryInfo.Region,
-		},
-		Repos: names,
+		Endpoint: endPoint,
+		Repos:    names,
 	}, logger)
 
 	images := make([]*RepoImgResp, 0)
 	if err == nil {
-		for _, repo := range repos.Repos {
-			for _, tag := range repo.Tags {
-				img := &RepoImgResp{
-					Host:  util.TrimURLScheme(registryInfo.RegAddr),
-					Owner: repo.Namespace,
-					Name:  repo.Name,
-					Tag:   tag,
+		resultChan := make(chan []*RepoImgResp, 1)
+		errChan := make(chan error, 1)
+		defer func() {
+			close(resultChan)
+			close(errChan)
+		}()
```

Comment:
suggest to close it by sender, otherwise may cause panic, if sender continue to write even it was closed.